test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,225 @@
+						Name:      "test-cm",
+						Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+					},
+				}
+				err := communityUserProxyClient.Create(context.TODO(), &cm)
+				require.Error(t, err)
+			})
+		})
+
+		/*
+		   Given Space exists for user A
+		   Given SSO user joe exists
+		   When  A flags their space visibility to "community"
+		   Then  joe can view A's Space
+		   And   joe cannot create resources in A's Space
+		*/
+		t.Run("as sso user", func(t *testing.T) {
+			// Given
+			userIdentity := &commonauth.Identity{
+				ID:       uuid.New(),
+				Username: "joe",
+			}
+			claims := []commonauth.ExtraClaim{commonauth.WithEmailClaim("joe@joe.joe")}
+			token, err := authsupport.NewTokenFromIdentity(userIdentity, claims...)
+			require.NoError(t, err)
+
+			joeCli, err := hostAwait.CreateAPIProxyClient(t, token, hostAwait.ProxyURLWithWorkspaceContext(sp.Name))
+			require.NoError(t, err)
+
+			t.Run("sso user can list config maps from space", func(t *testing.T) {
+				// then
+				cms := corev1.ConfigMapList{}
+				err := joeCli.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+				require.NoError(t, err)
+			})
+
+			t.Run("sso user cannot create config maps into space", func(t *testing.T) {
+				// then
+				cm := corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cm",
+						Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+					},
+				}
+				err := joeCli.Create(context.TODO(), &cm)
+				require.Error(t, err)
+			})
+		})
+	})
+
+	t.Run("space is not flagged as community", func(t *testing.T) {
+		sp := toolchainv1alpha1.Space{}
+		err = hostAwait.Client.Get(context.TODO(), client.ObjectKeyFromObject(space), &sp)
+		require.NoError(t, err)
+
+		/*
+		   Given Space exists for user A
+		   Given User community-user exists
+		   Given A space's visibility is NOT "community"
+		   Then  community-user cannot view A's Space
+		   And   community-user cannot create resources in A's Space
+		*/
+		t.Run("community user cannot access to non-community space", func(t *testing.T) {
+			require.NotEmpty(t, sp.Status.ProvisionedNamespaces)
+
+			t.Run("community user cannot list config maps from community space", func(t *testing.T) {
+				// then
+				cms := corev1.ConfigMapList{}
+
+				communityUserProxyClient, err := hostAwait.CreateAPIProxyClient(t, communityUser.token, hostAwait.ProxyURLWithWorkspaceContext(sp.Name))
+				require.NoError(t, err)
+
+				err = communityUserProxyClient.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+				require.Error(t, err)
+			})
+
+			t.Run("community user cannot create config maps into space", func(t *testing.T) {
+				cm := corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cm",
+						Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+					},
+				}
+				err := communityUserProxyClient.Create(context.TODO(), &cm)
+				require.Error(t, err)
```

Comment:
and here

Reply:
done in 8d2872804d812e52a62d4144e4451b21db20e798